source/develop/release-management/features/virt/dedicated-cpu.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Dedicated CPU Policies
+category: feature
+authors: tgolembi
+---
+
+# Dedicated CPU Policies
+
+## Summary
+
+This feature introduces policies for dynamically dedicating physical CPUs to
+the VM. While this can be (to some extent) done with static CPU pinning, the
+CPU pinning requires that the VM is pinned also to a particular host and that
+no other VMs (without pinning) are running on the host. Instead, this new
+feature allows the user to configure only the way how should physical CPUs be
+assigned to the VM and oVirt will take care of dynamically allocating suitable
+physical CPUs as well as evicting other VMs from those CPUs.
+
+
+## Owner
+
+* Name: Tomáš Golembiovský
+* Email: tgolembi@redhat.com
+
+
+## Description
+
+To be able to have more flexibility in configure how virtual CPUs (vCPUs)
+of VMs use physical CPUs (pCPUs) of the host this feature introduces several
+policies for dedicating CPUs. Compared to CPU pinning the advantages are that:
+
+  * VM does not need to be pinned to a specific host,
+  * solves some of the issues with *existing* pinning policy where
+    several VMs can end up to be pinned to same CPUs,
+  * *TODO* I am pretty sure there were couple more worth mentioning here
+
+The available policies described below divide the set of available pCPUs into
+three sub-sets, There are the dedicated CPUs that are assigned only to a single
+VM. Then there are some unassigned CPUs that cannot be used by any other VM
+because of restrictions requirements on the threads of the core. And finally
+there is the set of CPUs that are not strictly assigned to any particular VM
+and are shared between all the VMs with *shared* policy. This subset is hereby
+referred to as *shared pool*.
+
+The available CPU policies are:
+
+  * *shared*: *(default)* The guest vCPUs will be allowed to roam among all
+    pCPUs from shared pool.
+  * *dedicated*: The guest vCPUs will be strictly pinned to a set of host pCPUs
+    (similarly to static CPU pinning). The set of pCPUs will be chosen to match
+    the required guest CPU topology. If the host has SMT architecture, thread
+    siblings are preferred.
+  * *isolate-threads*: If the host does not have SMT architecture this is same
+    as *dedicated* policy. If the host has SMT architecture each vCPU is placed
+    on a different physical core and all thread siblings are excluded from use
+    by this and any other VM (rendering all but one thread from a core unusable).
+  * *require-threads*: The host must have an SMT architecture. Each vCPU is
```

Comment:
"require-threads" sounds a bit confusing. How about single-VM-threads or something else?

Reply:
Of course I am open to suggestions here and we can rename any of the policies if anyone has a good idea. But for this one I don't think `single-vm-threads` is an improvement.

Reply:
How about "consolidate-threads"? I still don't think there are any "required" threads here.

Reply:
What about just "siblings"?

Reply:
Better than "require-threads".

Reply:
I have changed it to `siblings` but I am not quite satisfied with it yet. If anyone still has some other suggestions feel free to share them.